### PR TITLE
Replace MockProcessManager with FakeProcessManager

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Switch to `FakeProcessManager` in a few places.

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart [hunk 1]
```diff
@@ -24,7 +24,7 @@ void main() {
   group('UpgradeCommandRunner', () {
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart:24]: This one is hard to read... good luck 😦

File: packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart [hunk 2]
```diff
@@ -144,96 +90,86 @@ void main() {
...
-      expect(
-        () async => await realCommandRunner.fetchRemoteRevision(),
+      processManager.addCommands(<FakeCommand>[
+        const FakeCommand(command: <String>[
+          'git', 'fetch', '--tags'
+        ]),
+        FakeCommand(command: const <String>[
+          'git', 'rev-parse', '--verify', '@{u}',
+        ],
+            onRun: () {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart:128]: formatting is kind of weird here, could you align the FakeCommand args?

Comment (jmagman) [reply packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart]: Done.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**